addons/ui_units/functions/fnc_onAltLoad.sqf

```diff
@@ -0,0 +1,38 @@
+#include "script_component.hpp"
+
+params ["_ctrl"];
+
+private _alt = (ctrlParent _ctrl) ctrlCreate [QGVAR(alt), -1];
+
+private _units = (vehicle ACE_player) call FUNC(altUnits);
+_units call FUNC(altInfo) params ["_unitText", "_ratio"];
+
+((ctrlParent _alt) displayCtrl 1005) ctrlSetText _unitText;
+GVAR(altRatio) = _ratio;
+
+private _settingEH = ["CBA_SettingChanged", {
+    params ["_setting", "_value"];
+    _thisArgs params ["_settingName", "_alt"];
+    if (_setting isEqualTo _settingName) then {
+        [_value] call FUNC(altInfo) params ["_unitText", "_ratio"];
+        ((ctrlParent _alt) displayCtrl 1005) ctrlSetText _unitText;
+        GVAR(altRatio) = _ratio;
+    };
+}, [_units select 1, _alt]] call CBA_fnc_addEventHandlerArgs;
+
+[{
+    (_this select 0) params ["_ctrl", "_alt", "_settingEH"];
+    if (isNull _ctrl) exitWith {
+        [_this select 1] call CBA_fnc_removePerFrameHandler;
+        ["CBA_SettingChanged", _settingEH] call CBA_fnc_removeEventHandler;
+    };
+    _ctrl ctrlSetFade 1;
+    _ctrl ctrlCommit 0;
+    private _altValue = (((getPosATL vehicle ACE_player) select 2) / GVAR(altRatio));
```

Comment:
ATL is based on bottom of the sea height above water. Shouldn't it be sea level = 0 instead?
I don't suggest AGL, because that changes with the waves (so it is dependant on time), but surely this needs to be handled differently over the sea, right?

Reply:
Hm, yeah that would be different than vanilla behaviour. Is there a way to determine if you are above water? I'm not familiar with one. It should switch to ASL in that case.

Reply:
ATL above land would make sense. Predict height of when you have to open the chute. No idea what BI does, why don't we just read their value with `parseNumber ctrlText` from the hard coded control and convert that if need to our units?

Reply:
We lose precision that way since we will be doing the conversions on rounded numbers. I'd like to ideally use the raw values so we get accurate results.

Reply:
Wait, couldn't any negative values just use 0?

Reply:
No.

Reply:
`(getPosATL player) - (0 min (getTerrainHeightASL getPos player))` should work just fine

Reply:
Fixed in 8ee50719e5be1d47c7a27d16dc707d0047126973